docroot/modules/humsci/hs_layouts/src/Plugin/Validation/Constraint/UniqueMainContentConstraintValidator.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace Drupal\hs_layouts\Plugin\Validation\Constraint;
+
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Entity\EntityDisplayRepository;
+use Drupal\Core\Plugin\Context\Context;
+use Drupal\Core\Plugin\Context\ContextDefinition;
+use Drupal\Core\Plugin\Context\EntityContext;
+use Drupal\layout_builder\LayoutTempstoreRepository;
+use Drupal\layout_builder\SectionStorage\SectionStorageManager;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Validates the hs_layouts_unique_main_content constraint.
+ */
+class UniqueMainContentConstraintValidator extends ConstraintValidator implements ContainerInjectionInterface {
+
+  /**
+   * Section Storage Manager service.
+   *
+   * @var \Drupal\layout_builder\SectionStorage\SectionStorageManager
+   */
+  private $storageManager;
+
+  /**
+   * Layout Tempstorage Repository service.
+   *
+   * @var \Drupal\layout_builder\LayoutTempstoreRepository
+   */
+  private $tempstoreRepository;
+
+  /**
+   * Layout Entity Display Repository service.
+   *
+   * @var \Drupal\Core\Entity\EntityDisplayRepository
+   */
+  private $displayRepository;
+
+  /**
+   * Create a new UniqueMainContentConstraintValidator instance.
+   *
+   * @param \Drupal\layout_builder\SectionStorage\SectionStorageManager $storageManager
+   *   Section storage manager service.
+   * @param \Drupal\layout_builder\LayoutTempstoreRepository $tempstoreRepository
+   *   Layout Tempstore Repository service.
+   * @param \Drupal\Core\Entity\EntityDisplayRepository $displayRepository
+   *   Layout Tempstore Repository service.
+   */
+  public function __construct(SectionStorageManager $storageManager, LayoutTempstoreRepository $tempstoreRepository, EntityDisplayRepository $displayRepository) {
+    $this->storageManager = $storageManager;
+    $this->tempstoreRepository = $tempstoreRepository;
+    $this->displayRepository = $displayRepository;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('plugin.manager.layout_builder.section_storage'),
+      $container->get('layout_builder.tempstore_repository'),
+      $container->get('entity_display.repository')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($node, Constraint $constraint) {
+    // Layout builder should be enabled and the layout should be overridable for
+    // the current node, otherwise return.
+    $view_display = $this->displayRepository->getViewDisplay('node', $node->bundle());
+    if (!($view_display->isLayoutBuilderEnabled() && $view_display->isOverridable())) {
```

Comment:
check if these methods even exist by making sure the object is an instance of `\Drupal\layout_builder\Entity\LayoutEntityDisplayInterface`